From 2717baf34693fc9aa5fa7d1f2a5e8eb1677c1bc2 Mon Sep 17 00:00:00 2001
From: Gavin Shan <gshan@redhat.com>
Date: Wed, 8 Jul 2020 10:02:25 +0200
Subject: hw/arm: Changes to rhel820 machine

RH-Author: Gavin Shan <gshan@redhat.com>
Message-id: <20200630013648.101937-1-gshan@redhat.com>
Patchwork-id: 97844
O-Subject: [RHEL-AV 8.3.0 qemu-kvm PATCH v5 1/3] hw/arm: Changes to rhel820 machine
Bugzilla: 1818843

This applies two changes to rhel820 machine:

   * Set the gic version to VIRT_GIC_VERSION_NOSEL by default, which
     doesn't cause functional changes.
   * Disallow to configure the RAS property, which is hidden by default.

Signed-off-by: Gavin Shan <gshan@redhat.com>
RH-Acked-by: Auger Eric <eric.auger@redhat.com>
RH-Acked-by: Andrew Jones <drjones@redhat.com>
Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 hw/arm/virt.c | 17 ++++++-----------
 1 file changed, 6 insertions(+), 11 deletions(-)

diff --git a/hw/arm/virt.c b/hw/arm/virt.c
index 5a45677..53f02e0 100644
--- a/hw/arm/virt.c
+++ b/hw/arm/virt.c
@@ -2090,6 +2090,7 @@ static void virt_set_acpi(Object *obj, Visitor *v, const char *name,
     visit_type_OnOffAuto(v, name, &vms->acpi, errp);
 }
 
+#if 0 /* Disabled for Red Hat Enterprise Linux */
 static bool virt_get_ras(Object *obj, Error **errp)
 {
     VirtMachineState *vms = VIRT_MACHINE(obj);
@@ -2103,6 +2104,7 @@ static void virt_set_ras(Object *obj, bool value, Error **errp)
 
     vms->ras = value;
 }
+#endif /* disabled for RHEL */
 
 static char *virt_get_gic_version(Object *obj, Error **errp)
 {
@@ -2760,12 +2762,8 @@ static void rhel820_virt_instance_init(Object *obj)
     object_property_set_description(obj, "highmem",
                                     "Set on/off to enable/disable using "
                                     "physical address space above 32 bits");
-    /*
-     * Default GIC type is still v2, but became configurable for RHEL. We
-     * keep v2 instead of max as TCG CI test cases require an MSI controller
-     * and there is no userspace ITS MSI emulation available.
-     */
-    vms->gic_version = 2;
+
+    vms->gic_version = VIRT_GIC_VERSION_NOSEL;
     object_property_add_str(obj, "gic-version", virt_get_gic_version,
                         virt_set_gic_version);
     object_property_set_description(obj, "gic-version",
@@ -2792,12 +2790,9 @@ static void rhel820_virt_instance_init(Object *obj)
     object_property_set_description(obj, "iommu",
                                     "Set the IOMMU type. "
                                     "Valid values are none and smmuv3");
+
     vms->ras = false;
-    object_property_add_bool(obj, "ras", virt_get_ras,
-                             virt_set_ras);
-    object_property_set_description(obj, "ras",
-                                    "Set on/off to enable/disable reporting host memory errors "
-                                    "to a KVM guest using ACPI and guest external abort exceptions");
+
     vms->irqmap=a15irqmap;
     virt_flash_create(vms);
 }
-- 
1.8.3.1

